tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -225,9 +238,20 @@ class D3D12Device : public RendererBase
             return (DeviceAddress)m_resource.getResource()->GetGPUVirtualAddress();
         }
 
-        virtual SLANG_NO_THROW Result SLANG_MCALL getNativeHandle(NativeHandle* outHandle) override
+        virtual SLANG_NO_THROW Result SLANG_MCALL getNativeResourceHandle(NativeResourceHandle* outHandle) override
+        {   
+            outHandle->handle = (uint64_t)m_resource.getResource();
+            outHandle->type = NativeHandleType::D3D12;
+            return SLANG_OK;
+        }
+
+        virtual SLANG_NO_THROW Result SLANG_MCALL getSharedHandle(SharedHandle* outHandle) override
         {
-            *outHandle = (uint64_t)m_resource.getResource();
+            ComPtr<ID3D12Device> pDevice;
+            auto pResource = m_resource.getResource();
+            pResource->GetDevice(IID_PPV_ARGS(pDevice.writeRef()));
+            SLANG_RETURN_ON_FAIL(pDevice->CreateSharedHandle(pResource, NULL, GENERIC_ALL, (L"Buffer" + std::to_wstring(id)).c_str(), outHandle));
+            id++;
```

Comment:
I think we can just make `id` a static local variable in this function instead of a static member. Just declare `static uint32_t id = 0;` in the function.

Reply:
> we can just pass in nullptr for the name argument and it will create an unnamed shared handle, with no chance of collision with other handles for other allocations

I think I added it initially to ensure unique names that would also indicate if a resource was a buffer or texture. But if that's the case and I also make the changes I mention above to `SharedHandle`, then yeah this won't be needed at all.

Reply:
Let's get rid of the string name generation logic here.